service/client/client.js

```diff
@@ -717,6 +717,14 @@ export default class Client {
             data = await response.json();
         } else {
             data = await response.text();
+            if (isResponseInvalid(data)) {
```

Comment:
I don't think I follow the logic here. It's not a JSON request so why are we checking if it's returning JSON to tell if it's valid?